pkg/occlient/templates.go

```diff
@@ -139,6 +139,51 @@ func generateSupervisordDeploymentConfig(commonObjectMeta metav1.ObjectMeta, com
 	return dc
 }
 
+func updateSupervisorDeploymentConfig(existingDc *appsv1.DeploymentConfig, commonObjectMeta metav1.ObjectMeta, commonImageMeta CommonImageMeta,
+	envVar []corev1.EnvVar, envFrom []corev1.EnvFromSource, resourceRequirements *corev1.ResourceRequirements) appsv1.DeploymentConfig {
+
+	dc := *existingDc
+
+	dc.ObjectMeta = commonObjectMeta
+	dc.Spec.Strategy.Type = appsv1.DeploymentStrategyTypeRecreate
+
+	dc.Spec.Selector = map[string]string{
+		"deploymentconfig": commonObjectMeta.Name,
+	}
+
+	dc.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+		Labels: map[string]string{
+			"deploymentconfig": commonObjectMeta.Name,
+		},
+		// https://github.com/openshift/odo/pull/622#issuecomment-413410736
+		Annotations: map[string]string{
+			"alpha.image.policy.openshift.io/resolve-names": "*",
+		},
+	}
+
+	for i := range dc.Spec.Template.Spec.Containers {
```

Comment:
all the containers would be the same?

Reply:
We have only one container support so I guess we can keep them same for now

Reply:
then why not directly go for the first container? This will make it really confusing when someones add another container. Going just for the first one would be much safer